handlers/chat.js

```diff
@@ -56,44 +54,95 @@ Dota2.Dota2Client.prototype.sendMessage = function(channel, message) {
+  var channelId = this.chatChannels.filter(function (item) {if (item.channel_name == channel) return true; }).map(function (item) { return item.channel_id; })[0]
   if (channelId === undefined) {
     if (this.debug) util.log("Cannot send message to a channel you have not joined.");
     return;
   }
-  var payload = dota_gcmessages_client.CMsgDOTAChatMessage.serialize({
-    "channelId": channelId,
+  var payload = new Dota2.schema.CMsgDOTAChatMessage({
+    "channel_id": channelId,
     "text": message
   });
-
-  this._client.toGC(this._appid, (Dota2.EDOTAGCMsg.k_EMsgGCChatMessage | protoMask), payload);
+  this._protoBufHeader.msg = Dota2.EDOTAGCMsg.k_EMsgGCChatMessage;
+  this._gc.send(this._protoBufHeader,
+                payload.toBuffer()
+  );
 };
 
+Dota2.Dota2Client.prototype.requestChatChannels = function() {
+  /* Attempts to send a message to a chat channel. GC does not send a response. */
+  if (!this._gcReady) {
+    if (this.debug) util.log("GC not ready, please listen for the 'ready' event.");
+    return null;
+  }
+
+  if (this.debug) util.log("Requesting channel list");
+  var payload = new Dota2.schema.CMsgDOTARequestChatChannelList({
+  });
+  this._protoBufHeader.msg = Dota2.EDOTAGCMsg.k_EMsgGCRequestChatChannelList;
+  this._gc.send(this._protoBufHeader,
+                payload.toBuffer()
+  );
+};
 
 // Handlers
 
 var handlers = Dota2.Dota2Client.prototype._handlers;
 
-handlers[Dota2.EDOTAGCMsg.k_EMsgGCJoinChatChannelResponse] = function onJoinChatChannelResponse(message) {
+var onJoinChatChannelResponse = function onJoinChatChannelResponse(message) {
   /* Channel data after we sent k_EMsgGCJoinChatChannel */
-  var channelData = dota_gcmessages_client.CMsgDOTAJoinChatChannelResponse.parse(message);
+  var channelData = Dota2.schema.CMsgDOTAJoinChatChannelResponse.decode(message);
+  util.log("Chat channel "+channelData.channel_name+ " has "+channelData.members.length+" person(s) online");
   this.chatChannels.push(channelData);
 };
+handlers[Dota2.EDOTAGCMsg.k_EMsgGCJoinChatChannelResponse] = onJoinChatChannelResponse;
 
-handlers[Dota2.EDOTAGCMsg.k_EMsgGCChatMessage] = function onChatMessage(message) {
+var onChatMessage = function onChatMessage(message) {
   /* Chat channel message from another user. */
-  var chatData = dota_gcmessages_client.CMsgDOTAChatMessage.parse(message);
+  var chatData = Dota2.schema.CMsgDOTAChatMessage.decode(message);
+  if(this.debug) util.log("Received chat message from "+chatData.persona_name+" in "+chatData.channel_id);
   this.emit("chatMessage",
-    this.chatChannels.filter(function (item) {if (item.channelId === chatData.channelId) return true; }).map(function (item) { return item.channelName; })[0],
-    chatData.personaName,
+    this.chatChannels.filter(function (item) {if (""+item.channel_id === ""+chatData.channel_id) return true; }).map(function (item) { return item.channel_name; })[0],
```

Comment:
If anyone has any suggestions on how to make the equality in the filter work without resorting to these shenanigans, feel free to let me know because this bugs the crap out me